papyri/gen.py

```diff
@@ -102,7 +102,8 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 self.log.exception("Unexpected error")
             if not self.config.early_error:
                 return True
-
+            if not self._expected_unseen and self.config.fail_unseen_error:
+                raise UnseenError()
```

Comment:
I think this needs to be dedented, as otherwise it should fail is there is no error and you expect one and it currently is in the `if exc_type` block.
```suggestion
        if not self._expected_unseen and self.config.fail_unseen_error:
            raise UnseenError()
```

Reply:
I think here `_expected_unseen.get(self._qa)` is necessary to extract the expected error from the current object we are testing.